File: Tease AI/Services/NotifyUser.vb
Option Strict On
Option Infer Off
Imports System.Threading.Tasks
Imports TeaseAI.Common.Interfaces

Public Class NotifyUser
    Implements INotifyUser

    Public Sub ModalMessage(message As String) Implements INotifyUser.ModalMessage
        MessageBox.Show(message, "Information", MessageBoxButtons.OK, MessageBoxIcon.Information)
    End Sub

    Public Function ModalMessageAsync(message As String) As Task Implements INotifyUser.ModalMessageAsync
        Throw New NotImplementedException()
    End Function
End Class

File: TeaseAI.PersonalityEditor.Uwp/Services/NotifyUser.cs
using System;
using System.Threading.Tasks;
using TeaseAI.Common.Interfaces;
using Windows.UI.Popups;

namespace TeaseAI.PersonalityEditor.Services
{
    public class NotifyUser : INotifyUser
    {
        public void ModalMessage(string message)
        {
            throw new NotImplementedException();
        }

        public async Task ModalMessageAsync(string message)
        {
            var dialog = new MessageDialog(message);
            await dialog.ShowAsync();
        }
    }
}
